go/array_string/min_window/solution.go

```diff
@@ -0,0 +1,101 @@
+// package main has the solution for the minimum window
+// algorithm implementation using the sliding window technique.
+package main
+
+import (
+	"fmt"
+	"math"
+	"strings"
+)
+
+func minWindow(mainStr, subStr string) string {
+	// validation
+	if len(mainStr) == 0 || len(subStr) == 0 || len(subStr) > len(mainStr) {
+		return ""
+	}
+	// using the sliding window technique.
+	// create hashmap for required and desired character.
+	required := make(map[string]int)
+	desired := make(map[string]int)
+	// ensure it is the count of the unique keys in desired
+	// and not the subStr incase of repeated characters.
+	substrCount := 0
+
+	// populate the desired and required hashmaps
+	for _, char := range subStr {
+		desired[string(char)] = 0
+
+		// create the new hashmap with the count of the required
+		// characters in the subStr.
+		if _, ok := required[string(char)]; !ok {
+			// if the character has not been accounted of.
+			substrCount++ // count the unique characters
+			required[string(char)] = strings.Count(subStr, string(char))
+		}
+	}
+
+	var slowPointer, fastPointer uint64
+	minWindowRange := []uint64{0, math.MaxUint64}
```

Comment:
Rather just use `int` so that it works nicely with `len()` without having to convert types